projects/extension/src/background/ConnectionManager.ts

```diff
@@ -188,31 +188,53 @@ export class ConnectionManager extends (EventEmitter as { new(): StateEmitter })
    * @param name - Name of the chain
    * @param spec - ChainSpec of chain to be added
    * @param jsonRpcCallback - The jsonRpcCallback function that should be triggered
+   * @param relayChain - optional SmoldotChain for relay chain
    * 
    * @returns addedChain - An the newly added chain info
    */
   async addChain(
-    name: string,
     chainSpec: string,
-    jsonRpcCallback: SmoldotJsonRpcCallback): Promise<SmoldotChain> {
+    jsonRpcCallback?: SmoldotJsonRpcCallback,
+    relayChain?: SmoldotChain): Promise<SmoldotChain> {
     if (!this.#client) {
       throw new Error('Smoldot client does not exist.');
     }
 
-    const addedChain = await this.#client.addChain({
-      chainSpec,
-      jsonRpcCallback,
-      potentialRelayChains: this.#networks.map(net => net.chain),
-    });
-
-    this.#networks.push({
-      name,
-      chain: addedChain,
-      status: 'connected',
-      isKnown: true,
-      chainspecPath: `${name}.json`
-    });
-
-    return addedChain;
+    if (relayChain) {
+      const addedChain = await this.#client.addChain({
+        chainSpec,
+        jsonRpcCallback,
+        potentialRelayChains: [relayChain],
```

Comment:
Again, this should be all chains that have been previously added by the same uApp. There's no need for a `relayChain` parameter.